### PR TITLE
Restore HTTP error handling for haalcentraal/bag + further refactor of Haal Centraal client

### PR DESCRIPTION
**Pull request by lbam**
See commit messages for details.

### CODE REVIEW

File: src/dso_api/dynamic_api/remote/clients.py [hunk 1]
```diff
@@ -42,6 +42,109 @@ class RemoteResponse:
...
+        response.status,
+        response.reason,
+        (time.perf_counter_ns() - t0) * 1e-9,
+    )
+
+    if response.status >= 200 and response.status < 300:
+        return response
+
+    # We got an error.
+    if logger.isEnabledFor(logging.DEBUG):
```
Comment (jjmurre) [inline src/dso_api/dynamic_api/remote/clients.py:83]: Is the `if` necessary? Because the logger handles this for you depending on the log level.

Comment (lbam) [reply src/dso_api/dynamic_api/remote/clients.py:83]: This is ancient code from 2020, see ab346a13764c31e6b804f3b8d11708708b079f31. It looks overly cautious to me too, I'll see if I can get rid of it.

Comment (lbam) [reply src/dso_api/dynamic_api/remote/clients.py:83]: I think the issue is that `.data` is a `property` that parses and caches the response, which we don't want to do except if we actually want to print the message.

Comment (jjmurre) [reply src/dso_api/dynamic_api/remote/clients.py:83]: Ah, makes sense.

**Review by jjmurre — APPROVED** [review]
LGTM

**Pull request merged**